ui-lib/src/components/Layout/SideBar/components/SideBarItem/SideBarItem.tsx

```diff
@@ -0,0 +1,109 @@
+import React, { useContext } from "react";
+import styled from "styled-components";
+import Text from "@ui/components/Text";
+import { SideBarContext } from "@ui/components/Layout/SideBar/SideBar";
+import TransitionInOut from "@ui/components/Transition/TransitionInOut";
+
+const Item = styled.li`
+  /** DEFAULT VARIANT **/
+  --ll-sidebar-item-label-color: ${props => props.theme.colors.palette.v2.text.secondary};
+  --ll-sidebar-item-icon-color: ${props => props.theme.colors.palette.v2.text.secondary};
+  --ll-sidebar-item-background-color: unset;
+
+  display: flex;
+  flex-direction: row;
+  align-items: center;
+  column-gap: 0.875rem;
+  padding: 0.875rem 1rem;
+  border-radius: 8px;
+  min-height: 52px;
+
+  color: var(--ll-sidebar-item-icon-color);
+  background-color: var(--ll-sidebar-item-background-color);
+  cursor: pointer;
+
+  /** HOVER VARIANT **/
+  &:hover {
+    --ll-sidebar-item-label-color: ${props => props.theme.colors.palette.v2.text.default};
+    --ll-sidebar-item-icon-color: ${props => props.theme.colors.palette.v2.primary.dark};
+    --ll-sidebar-item-background-color: unset;
+  }
+
+  /** FOCUS VARIANT **/
+  &:focus {
+    box-shadow: 0px 0px 0px 4px rgba(187, 176, 255, 0.4);
+    border-radius: 8px;
+  }
+
+  /** ACTIVE VARIANT **/
+  &[data-active] {
+    --ll-sidebar-item-label-color: ${props => props.theme.colors.palette.v2.text.default};
+    --ll-sidebar-item-icon-color: ${props => props.theme.colors.palette.v2.primary.dark};
+    --ll-sidebar-item-background-color: ${props =>
+      props.theme.colors.palette.v2.primary.backgroundLight};
+  }
+
+  /** DISABLE VARIANT **/
+  &[data-disable] {
+    --ll-sidebar-item-label-color: ${props => props.theme.colors.palette.v2.text.secondary};
+    --ll-sidebar-item-icon-color: ${props => props.theme.colors.palette.v2.text.secondary};
+    --ll-sidebar-item-background-color: unset;
+    opacity: 0.3;
+    cursor: unset;
+  }
+`;
+
+export const SideBarItemLabel = styled(Text)`
+  display: inline-block;
+  color: var(--ll-sidebar-item-label-color);
+
+  &:first-letter {
+    text-transform: uppercase;
+  }
+`;
+
+export type SideBarItemType = {
+  label: string;
+  children: JSX.Element;
+  onClick: () => void;
+  isActive?: boolean;
+  isDisable?: boolean;
+};
+
+const SideBarItem = ({
+  label,
+  children,
+  onClick,
+  isActive,
+  isDisable,
+}: SideBarItemType): JSX.Element => {
+  const { isExpanded } = useContext(SideBarContext);
+
+  const handleClick = () => {
+    if (isDisable) return;
```

Comment:
nitpicking ⛏️ : can be reduced to `!isDisable && onClick();`

Reply:
yeaaah, maybe I'm old school but the two-line solution looks clearer for me. I'll keep it if you don't mind 👀

Reply:
sure no worries 👍